satpy/readers/lma_l2.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019 Satpy developers
+#
+# This file is part of Satpy.
+#
+# Satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# Satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# Satpy.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+"""
+ LMA flash extent 2d data
+"""
+
+import numpy as np
+import xarray as xr
+from satpy.readers.netcdf_utils import NetCDF4FileHandler, netCDF4
+from pyresample.geometry import AreaDefinition
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class LMAflashextent2dFileHandler(NetCDF4FileHandler):
+    """NetCDF4 reader for LMA L2 (FED-2d)."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Initialize the reader."""
+        super(LMAflashextent2dFileHandler, self).__init__(filename, filename_info, filetype_info)
+
+    def available_datasets(self, configured_datasets=None):
```

Comment:
This method is getting a bit long, can you create a couple of private methods to delegate some of the work?